### PR TITLE
Chore: Add support for incremental linked builds

### PR DESCRIPTION
**Pull request by jstoffan**
This change will support a "watch" mode for use during development via `yarn link`.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #613

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: .babelrc [hunk 1]
```diff
@@ -90,6 +90,37 @@
                     }
                 ]
             ]
+        },
+        "watch": {
+            "presets": [
```
Comment (priyajeet) [inline .babelrc]: file changed in other commit

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "build-npm": "yarn run npm && yarn run build-lib",
     "build-link": "yarn run npm && yarn run build-link-lib",
     "build-link-lib": "yarn run copy-lib && BABEL_ENV=npm NODE_ENV=dev ./node_modules/.bin/babel lib -d lib -s inline && yarn run copy-messages",
+    "build-link-watch": "BABEL_ENV=watch NODE_ENV=dev babel src --out-dir lib --copy-files --ignore \"**/__tests__/**\" -s inline --watch",
```
Comment (priyajeet) [inline package.json]: maybe start using this style `build:link-watch`?
I'll change others soon.

File: .babelrc (hunk 1 above)
Comment (jstoffan) [reply .babelrc]: Yeah, I'm going to merge the babel upgrade first, then update this PR as needed.

File: package.json [hunk 3]
```diff
@@ -64,6 +65,7 @@
     "minor": "./build/release.sh -n && ./build/publish.sh && ./build/publish-examples.sh",
     "patch": "./build/release.sh -p && ./build/publish.sh && ./build/publish-examples.sh",
     "pre-build": "yarn install && npm-run-all clean build:i18n",
+    "prebuild-link-watch": "yarn install && rm -rf lib",
```
Comment (alexkrolick) [inline package.json]: I got it working on my branch using this:

```json
{
    "prebuild-link-dev": "npm-run-all -s build-locales copy-messages",
    "build-link-dev": "BABEL_ENV=dev NODE_ENV=dev babel src --out-dir lib --copy-files --ignore \"**/__tests__/**,src/components/messages.js\" -s inline --watch",
    "build-locales": "mkdir -p lib/components && node ./build/props-to-es2015.js"
}

**Pull request merged**